Homework6: Anton Stychnevsky

Stychnevsky opened the pull request:
Homework # 6

Stychnevsky committed:
Homework # 6

Stychnevsky committed:
fixed according PEP8

Stychnevsky committed:
Add docstrings

Stychnevsky committed:
fixed according PEP8

del space in start of docstrings

belovmd commented on HW6.py:
```diff
@@ -0,0 +1,224 @@
+    """Product is some food we can eat or put to refrigerator
+
+    It can be cold and freeze type.
+    Every product aso have size -oit shows how many cells this product need
```
it 
misprint

belovmd commented on HW6.py:
```diff
@@ -0,0 +1,224 @@
+
+    def advertisement(self):
+        """Advertising of master. Skills always increase by two"""
+        print('Highly skilled master {0} from {1} company is looking for work!'
```
i would prefer key value args here

belovmd replied:
i would prefer key value args here

belovmd commented on HW6.py:
```diff
@@ -0,0 +1,224 @@
+class RefrigeratorAtlant(Refrigerator):
+    """Belorussian refrigerator with small capacity"""
+    made_in = 'Belarus'
+    freeze_cam_capacity = 4
```
i don't understand why it doesn't depend on model of fridge

Stychnevsky replied:
Fixed.
Now every company has own models row. And we can change model or create new using "create_or_update_model" class method

belovmd commented on HW6.py:
```diff
@@ -0,0 +1,224 @@
+
+    New refrigerator always is empty and able to work
+    """
+    freeze_cells = 0
```
Am I correct if we'll have to fridges they would have same cells num

Stychnevsky replied:
Yes, because zero mean that it is zero occupied cells in new refrigerator.

I rename this variable to "occupied_cells" to make it clearer

belovmd commented on HW6.py:
```diff
@@ -0,0 +1,224 @@
+
+class Child(User):
+    """Child is user who can hit refrigerator"""
+    def hit_fridge(self, refr):
```
maybe it should be added to get or put food method for child?

Stychnevsky replied:
But child have "get" and "put" methods, because class Child is inheritor of class User, and class User have those methods

belovmd replied:
we can reload them, call super and add private _hit method to them I meant

belovmd commented on HW6.py:
```diff
@@ -0,0 +1,224 @@
+    It can be cold and freeze type.
+    Every product aso have size -oit shows how many cells this product need
+    """
+    in_cold_cam = None
```
When I'll put products would be created instance variable for each instance. 
the only use of these vars are not to take exception in your if-elif condition. it should be refactored.

Stychnevsky replied:
Fixed.
Now instead of "in_cold_cam" and "in_freeze_cam" it is only one product property - location.

belovmd commented on HW6.py:
```diff
@@ -0,0 +1,224 @@
+            if self.skill >= 3:
+                refr.broken = False
+                customer.money -= self.skill * 100 * 1
+                self.skill += 0.5
```
he can unlimited improve his skill repairing atlants and can't level up when repair netherlands.

Stychnevsky replied:
When I invented I mean that Belorussian refrigerators are so bad, that every time they have new problem and master can improve his skills solving it :)
But I agree that is is not logical. Now fixed: Netherlands refrigerators also help to level up. And skills limit has been set.

belovmd replied:
I have new Atlant fridge and really like it ^)

belovmd reviewed (commented):
overall it's too many elif as for me. 
also please take a look at these issues:

Stychnevsky commented:
> overall it's too many elif as for me.

Fixed. Now it is no elifs in program. I did it using:
1) Dicts instead of two property variables, for example: capacity = {'freeze': 4,'cold': 6} instead of freeze_cam_capacity = 4 and cold_camp_capacity = 6
2) Abstract class Refrigerator
3) In first version of program "cold" products had to be put in freeze camera if it is no place in cold camera. This fact highly complicated the logic of the model and is not very close to life. So, I decided to make it easier. Now "cold" products can be put only in cold camera,  and "freeze" product only in freeze camera

Stychnevsky committed:
refactored and improved

- now cold product can be put only in cold camera
- take_product method do not have refr argument now
- create dicts instead of two similar variables
- master will level up after fixing all types of refrigerators
- limit of masters skill created
- model rows of every refrigerators class has been created
- every model can be updated (like new version in real life)
- refactor program to avoid elifs
- refactor to key value args in print() functions

Stychnevsky committed:
fixed according PEP8

belovmd reviewed (commented):
8,5
